test/index.js

```diff
@@ -0,0 +1,144 @@
+        it('should return false when u0338 is present', () => {
+            assert.equal(validator.validate('\u0338'), false);
+        })
+        it('should return false when u2027 is present', () => {
+            assert.equal(validator.validate('\u2027'), false);
+        })
+        // Han (CJK Ideographs) can be mixed with Bopomofo
+        it('should return true when han is mixed with bopomofo', () => {
+            assert.equal(validator.validate('\u4e00\u311B'), true);
+        })
+        // Han can be mixed with Hiragana
+        it('should return true when han is mixed with hiragana', () => {
+            assert.equal(validator.validate('\u4e00\u3077'), true);
+        })
+        // Han can be mixed with Katakana
+        it('should return true when han is mixed with katakana', () => {
+            assert.equal(validator.validate('\u4e00\u30DB'), true);
+        })
+        // Han can be mixed with Korean Hangul
+        it('should return true when han is mixed with hangul', () => {
+            assert.equal(validator.validate('\u4e00\u1100'), true);
+        })
+        it('should return false when deviation characters are present', () => {
+            assert.equal(validator.validate('\u00df\u03c2\u200c\u200d'), false);
+        })
+        it('should return true when input is ascii', () => {
+            assert.equal(validator.validate('ethereum.eth'), true);
+        })
+        // Latin, Cyrillic or Greek characters cannot be mixed with each other
+        it('should return false when mixed scripts are present', () => {
+            assert.equal(validator.validate('\u0000\u0100\u0370\u04009\u0300'), false);
+        })
+        it('should return false when cyrillic characters, which look like latin, are present', () => {
+            assert.equal(validator.validate('асԁеһіјӏорԛѕԝхуъЬҽпгѵѡ'), false);
+        })
+        it('should return true when single numerics are present', () => {
+            assert.equal(validator.validate('123456789'), true);
+        })
+        it('should return false when mixed numerics are present', () => {
+            assert.equal(validator.validate('1১'), false);
+            assert.equal(validator.validate('01०१'), false);
+        })
+        it('should return false when label matches dangerous pattern', () => {
+            assert.equal(validator.validate('\u30fb'), false);
+        })
+        it('should return true when single combining mark is present', () => {
+            assert.equal(validator.validate('\u0369'), true);
+        })
+        it('should return false when sequence of combining marks are present', () => {
+            assert.equal(validator.validate('\u0369\u0369\u0369'), false);
+        })
+    })
+})
+
+const {
+    ascii,
+    deviation,
+    combiningDiacriticException,
+    cyrillic,
+    cyrillicLikeLatin,
+    greek,
+    kanaCharacterException,
+    latinGreekCyrillicAscii,
+    nonAsciiLatin
+} = require('../src/regexes')
+
+describe('regexes', () => {

```

Comment:
would prefer to have this in its own file so you can have seperation of concerns here